newm-server/src/main/kotlin/io/newm/server/features/minting/repo/MintingRepositoryImpl.kt

```diff
@@ -0,0 +1,650 @@
+
+            mapItemKey = "links".toPlutusData()
+            mapItemValue = plutusData {
+                map = plutusDataMap {
+                    with(mapItem) {
+                        if (websites.isNotEmpty()) {
+                            add(
+                                plutusDataMapItem {
+                                    mapItemKey = "website".toPlutusData()
+                                    mapItemValue = if (websites.size > 1) {
+                                        plutusData {
+                                            list = plutusDataList {
+                                                listItem.addAll(websites.map { it.toPlutusData() })
+                                            }
+                                        }
+                                    } else {
+                                        websites[0].toPlutusData()
+                                    }
+                                }
+                            )
+                        }
+                        if (instagrams.isNotEmpty()) {
+                            add(
+                                plutusDataMapItem {
+                                    mapItemKey = "instagram".toPlutusData()
+                                    mapItemValue = if (instagrams.size > 1) {
+                                        plutusData {
+                                            list = plutusDataList {
+                                                listItem.addAll(instagrams.map { it.toPlutusData() })
+                                            }
+                                        }
+                                    } else {
+                                        instagrams[0].toPlutusData()
+                                    }
+                                }
+                            )
+                        }
+                        if (twitters.isNotEmpty()) {
+                            add(
+                                plutusDataMapItem {
+                                    mapItemKey = "twitter".toPlutusData()
+                                    mapItemValue = if (twitters.size > 1) {
+                                        plutusData {
+                                            list = plutusDataList {
+                                                listItem.addAll(twitters.map { it.toPlutusData() })
+                                            }
+                                        }
+                                    } else {
+                                        twitters[0].toPlutusData()
+                                    }
+                                }
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    companion object {
+        // FIXME: these should come from the eveara api and we should flag our db with which ones are contributing roles
```

Comment:
This will be fixed once I implement the eveara api and fetch these roles on a schedule.